src/dscl.ts

```diff
@@ -0,0 +1,316 @@
+// https://stackoverflow.com/a/70994696/6183068
+const satisfies =
+  <T>() =>
+  <U extends T>(u: U) =>
+    u;
+
+const generateDsclPath: Fig.Generator = {
+  trigger: "/",
+  getQueryTerm: "/",
+  custom: async (tokens, executeShellCommand) => {
+    const lastToken = tokens[tokens.length - 1];
+    if (lastToken === "") {
+      return [];
+    }
+    // can't guarantee that dscl is the first token. If it's not found,
+    // this is -1, so searching for `datasource` will start from index 0
+    const dsclIndex = tokens.findIndex((token) => token === "dscl");
```

Comment:
Not sure why I went straight to `findIndex` the first time...
```suggestion
    const dsclIndex = tokens.indexOf("dscl");
```